src/com/yashu/interview/map/linkedhashset/HashSet.java

```diff
@@ -0,0 +1,218 @@
+		if(collision(value)){
+			linkedList.setTail(avoidCollision(value,newNode,linkedList.getTail(),indexOfString));
+			count--;
+			return;
+		}
+		insertIntoArray(indexOfString, newNode);
+		linkedList.insert(newNode);
+	}
+
+
+	Node<T>[] doubleArray(int length) {
+		@SuppressWarnings("unchecked")
+		Node<T>[] temp = new Node[length];
+		for(int i = 0;i<length;i++){
+			temp[i] = array[i];
+		}
+		createNewArray(2*length);
+		@SuppressWarnings("unchecked")
+		Node<T>[] hashSetArray = new Node[2*length];
+		for(int i =0;i<length;i++){
+			hashSetArray[i] = temp[i];
+		}
+		return hashSetArray;
+	}
+
+	Node<T> avoidCollision(T value, Node<T> newNode, Node<T> tail, int indexOfString) {
+		Node<T> temp = array[indexOfString];
+		while(temp.getRight()!=null){
+			temp = temp.getRight();
+		}
+		temp.setRight(newNode);
+		linkedList.getTail().setNext(newNode);
+		newNode.setPrevious(tail);
+		linkedList.setTail(linkedList.getTail().getNext());
+		return linkedList.getTail();
+	}
+
+	private boolean collision(T value) {
+		return array[hashFunction(value)]!=null;
+	}
+
+	 public boolean exists(T value) {
+		if(array[hashFunction(value)]==null){
+			return false;
+		}else{
+			Node<T> temp = array[hashFunction(value)];
+			while(temp!= null){
+				if(temp.getData().equals(value)){
+					return true;
+				}
+				temp =temp.getRight();
+			}
+			return false;
+		}
+	}
+
+	private void insertIntoArray(int indexOfString, Node<T> newNode) {
+		array[indexOfString] = newNode;
+	}
+
+	int hashFunction(T value) {
+		return indexCalculator.getIndex(value,array.length);
+	}
+	
+	public void print() {
+		Node<T> temp = linkedList.getHead();
+		if(linkedList.getHead() == null){
+			System.out.println("Empty list!");
+			return;
+		}else{ 
+		while(temp!=null){
+			System.out.println(temp.getData());
+			temp = temp.getNext();
+		}
+		}
+	}
+   //Returns true if set is empty
+	boolean isEmpty(){
+		return linkedList.getHead()==null;
+	}
+	/**
+	 * Removes specified element from the set if it is present.
+	 */
+	public void delete(T value) {
+		/**
+		 * If the element doesn't exists 
+		 * @throw InputMisMatchException().
+		 */
+		if(!exists(value)){
+			throw new InputMismatchException();
+		}else{
+			int index = hashFunction(value);
+			Node<T> element = array[index];
+			if(isCollisionLinkedList(element)){
+				Node<T> newNode = deleteLinkedList(element,value);
+				array[index] = newNode;
+			}else{
+				array[index] = null;
+				linkedList.delete(element);
+			} 
+		}
+				 
+	}
+
+	private Node<T> deleteLinkedList(Node<T> element,T value) {
+		if(element.getData().equals(value)){
+			linkedList.delete(element);
+			return element.getNext();
+		}else{
+			Node<T> previous = element;
+			Node<T> current = element;
+			while(current!=null){
+				if(current.getData().equals(value)){
+					 linkedList.delete(current);
+					previous.setRight(current.getRight());
+				}
+				previous = current;
+				current = current.getNext();
```

Comment:
Shouldn't you stop looping after you found the element? 

This whole linked list manipulation is more confusing now. Some code is here and some is in LinkedList class.
Lets discuss on phone about how to improve this.